src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php

```diff
@@ -37,16 +37,17 @@ protected function createController($controller)
     {
         if (false === strpos($controller, '::') && 2 === substr_count($controller, ':')) {
             // controller in the a:b:c notation then
-            $controller = $this->parser->parse($controller);
-        }
-
-        $resolvedController = parent::createController($controller);
+            $deprecatedNotation = $controller;
+            $controller = $this->parser->parse($deprecatedNotation);
```

Comment:
will give an extra notice about the deprecated class being used, right? I would change ControllerNameParser to trigger from its constructor only (muting the notice using an implicit constructor arg for internal uses)

Reply:
or disable the notice in the method call

Reply:
done